src/Chapter2.hs

```diff
@@ -832,7 +873,16 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n aList
+  | n < 0 || length aList == 0 = []
+  | otherwise =
+      take listLength (drop n (take repeatCycle (cycle aList)))
+      where
+        listLength = length aList
+        repeatCycle = listLength * (n + 1)
```

Comment:
I don't quite understand this moment, could you please elaborate on your algorithm?